test/parallel/test-tls-cipher-list.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+var common = require('../common');
+
+if (!common.hasCrypto) {
+  console.log('1..0 # Skipped: missing crypto');
+  return;
+}
+
+var assert = require('assert');
+var spawn = require('child_process').spawn;
+const default_core_list = require('constants').defaultCoreCipherList;
+
+function doCheck(arg, env, check) {
+  var out = '';
+  var arg = arg.concat([
+    '-pe',
+    'require("constants").defaultCipherList'
+  ]);
+  spawn(process.execPath, arg, env).
+     on('error', assert.fail).
+    stdout.on('data', function(chunk) {
+      out += chunk;
+    }).on('end', function() {
+      assert.equal(out.trim(), check);
+    }).on('error', assert.fail);
+}
+
+// test the default unmodified version
+doCheck([], {}, default_core_list);
+
+// test the command line switch by itself
+doCheck(['--cipher-list=ABC'], {}, 'ABC');
+
+// test the environment variable by itself
+doCheck([], {env:{NODE_CIPHER_LIST:'ABC'}}, 'ABC');
+
+// test that the environment variable overrides the command line switch
+doCheck(['--cipher-list=ABC'], {env:{NODE_CIPHER_LIST:'XYZ'}}, 'XYZ');
```

Comment:
Aren't we throwing any proper error if the cipher is not a valid one? I think it will be better if we capture that as well in the test. What do you think?